File: src/main/java/com/morak/performancetracker/context/ContextManager.java
package com.morak.performancetracker.context;

public interface ContextManager {

    default void beforeClass(TestMetadata testMetadata) {
    }

    default void beforeEach(TestMetadata testMetadata) {
    }

    default void afterEach(Accumulator accumulator, TestMetadata testMetadata) {
    }

    default void afterClass(Accumulator accumulator, TestMetadata testMetadata) {
    }

    void afterAll(Accumulator accumulator);
}

File: src/main/kotlin/com/morak/performancetracker/context/ContextManager.kt
package com.morak.performancetracker.context

interface ContextManager {
    fun beforeClass(testMetadata: TestMetadata) {}
    fun beforeEach(testMetadata: TestMetadata) {}
    fun afterEach(accumulator: Accumulator, testMetadata: TestMetadata) {}
    fun afterClass(accumulator: Accumulator, testMetadata: TestMetadata) {}
    fun afterAll(accumulator: Accumulator)
}
